src/dll_api.cpp

```diff
@@ -414,4 +414,9 @@ WIN_SPARKLE_API void __cdecl win_sparkle_check_update_without_ui()
 }
 
 
+WIN_SPARKLE_API void __cdecl win_sparkle_set_zip_download_callback(void (*cb)(std::wstring))
```

Comment:
Contrary to the claim that this adds nothing to the API, you're doing just that here — but without adding it to the public header or documenting it.

Reply:
That is correct. I took a shortcut because I realized that adding a std::wstring dependency would break your C compatibility.

Reply:
Make the callback take `const wchar_t*` argument (obtained with `wxString::wc_str()`) — it's read only and NULL-terminated, there's no risk there.

Reply:
Roger that.